gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/SubscriptionServiceTest.java

```diff
@@ -860,6 +863,89 @@ public void shouldCloseSubscription() throws Exception {
             .trigger(eq(GraviteeContext.getExecutionContext()), eq(ApplicationHook.SUBSCRIPTION_CLOSED), nullable(String.class), anyMap());
     }
 
+    @Test
+    public void shouldFailSubscription() throws TechnicalException {
+        Subscription subscription = buildTestSubscription(ACCEPTED);
+        subscription.setId(SUBSCRIPTION_ID);
+        final long yesterday = Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli();
+        final Date initialUpdateDate = new Date(yesterday);
+        subscription.setUpdatedAt(initialUpdateDate);
+
+        when(subscriptionRepository.findById(SUBSCRIPTION_ID)).thenReturn(Optional.of(subscription));
+        when(subscriptionRepository.update(subscription)).thenReturn(subscription);
+
+        final String failureCause = "💥 Endpoint not available";
+        subscriptionService.fail(SUBSCRIPTION_ID, failureCause);
+
+        verify(subscriptionRepository).findById(SUBSCRIPTION_ID);
+        ArgumentCaptor<Subscription> subscriptionCaptor = ArgumentCaptor.forClass(Subscription.class);
+        verify(subscriptionRepository).update(subscriptionCaptor.capture());
+
+        final Subscription subscriptionCaptured = subscriptionCaptor.getValue();
+        assertThat(subscriptionCaptured.getConsumerPausedAt()).isNull();
+        assertThat(subscriptionCaptured.getConsumerStatus()).isEqualTo(Subscription.ConsumerStatus.FAILURE);
+        assertThat(subscriptionCaptured.getFailureCause()).isEqualTo(failureCause);
+        assertThat(subscriptionCaptured.getUpdatedAt()).isAfter(initialUpdateDate);
+    }
+
+    @Test(expected = SubscriptionNotFoundException.class)
+    public void shouldNotFailSubscriptionBecauseDoesNotExist() throws TechnicalException {
+        Subscription subscription = buildTestSubscription(ACCEPTED);
+        subscription.setId(SUBSCRIPTION_ID);
+        final Date initialUpdateDate = new Date();
+        subscription.setUpdatedAt(initialUpdateDate);
+
+        when(subscriptionRepository.findById(SUBSCRIPTION_ID)).thenReturn(Optional.empty());
+
+        subscriptionService.fail(SUBSCRIPTION_ID, "💥 Endpoint not available");
+
+        verify(subscriptionRepository, never()).update(any());
```

Comment:
This should not be executed because the previous line throw an exception 🤔 maybe we could use AssertJ to catch the error